src/actix/api/snapshot_api.rs

```diff
@@ -23,13 +24,26 @@ pub struct SnapshottingParam {
     pub wait: Option<bool>,
 }
 
-// Actix specific code
-pub async fn do_get_full_snapshot(toc: &TableOfContent, snapshot_name: &str) -> Result<NamedFile> {
-    let file_name = get_full_snapshot_path(toc, snapshot_name)
-        .await
-        .map_err(storage_into_actix_error)?;
-
-    Ok(NamedFile::open(file_name)?)
+pub async fn do_get_full_snapshot(
+    dispatcher: &Dispatcher,
+    snapshot_name: &str,
+    wait: bool,
+) -> Result<NamedFile> {
+    let dispatcher = dispatcher.clone();
+    let snapshot_name = snapshot_name.to_string();
+    let task =
+        tokio::spawn(async move { get_full_snapshot_path(dispatcher.toc(), &snapshot_name).await });
+
+    if wait {
+        let filename = task.await;
+        if let Err(e) = filename {
+            return Err(ErrorInternalServerError(e.to_string()));
+        }
+        let filename = filename.unwrap().map_err(storage_into_actix_error)?;
+        Ok(NamedFile::open(filename)?)
+    } else {
+        Ok(NamedFile::open("not_found")?)
```

Comment:
if the snapshot does not exist, we need a proper 404

Reply:
I am not sure this function can be `non waiting`.
The user wants to retrieve the path via a single API call.

Reply:
Yes, it makes sense. Not all of API in snapshot should have `wait` parm.